### PR TITLE
fix: various fixes to input validation in signup

### PR DESCRIPTION
**Pull request by suprko**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/13029/desktop-update-text-input-helper-error-messages-for-sign-up-in
desktop: https://github.com/PeerioTechnologies/peerio-desktop/pull/421

#### Testing instructions  
see desktop pr

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ebilg** [comment]
can be merged

File: src/helpers/validation/user-validators.js [hunk 1]
```diff
@@ -31,12 +31,13 @@
  */
 const socket = require('../../network/socket');
 const { getFirstLetter } = require('../string');
+const config = require('../../config');
 
 const VALIDATION_THROTTLING_PERIOD_MS = 400;
 const usernameRegex = /^\w{1,16}$/;
-const emailRegex = /^[^ ]+@[^ ]+/i;
+const emailRegex = /^[\w]+@[\w]+\.+[\w]+/i;
```
[the remark this answers is not in the document]
Comment (anri-asaturov) [reply src/helpers/validation/user-validators.js]: ehhh, email validation is an ungrateful thing, that's why we don't do it (doing very simple version if it).

No, your regex is not correct. It doesn't allow `.` and `+` for example.

https://stackoverflow.com/questions/2049502/what-characters-are-allowed-in-an-email-address

[the remark this answers is not in the document]
Comment (suprko) [reply src/helpers/validation/user-validators.js]: So what is the server doing that rejects an email that passes emailRegex? Is it also a regex based validation, or something else?

[the remark this answers is not in the document]
Comment (anri-asaturov) [reply src/helpers/validation/user-validators.js]: My suggestion would be - leave the regex as is, change the error message to more generic one, OR make server return separate errors for invalid and taken emails.

[the remark this answers is not in the document]
Comment (anri-asaturov) [reply src/helpers/validation/user-validators.js]: Server validates email with Joi i think

[the remark this answers is not in the document]
Comment (anri-asaturov) [reply src/helpers/validation/user-validators.js]: Debug it, check what server returns in different cases.

**Pull request merged**